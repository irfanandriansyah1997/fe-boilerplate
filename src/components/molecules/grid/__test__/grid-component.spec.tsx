import { render } from '@testing-library/react';

import Grid from '..';

describe(`Testing Grid Component`, () => {
  it(`Snapshot Testing`, () => {
    const { queryAllByTestId } = render(
      <Grid Cell={() => <div>Cell</div>} row={3} column={4} />
    );
    const container = queryAllByTestId(/grid-container/);

    expect(container).toMatchSnapshot();
  });

  it(`Testing Render Grid Component`, () => {
    const { queryAllByTestId } = render(
      <Grid Cell={() => <div>Cell</div>} row={3} column={4} />
    );
    const item = queryAllByTestId(/grid-item/);
    const row = queryAllByTestId(/grid-row/);
    const [firstRow] = row;

    expect(row.length).toBe(3);
    expect(firstRow.querySelectorAll(`[data-testid='grid-item']`).length).toBe(
      4
    );
    expect(item.length).toBe(12);
  });
});
