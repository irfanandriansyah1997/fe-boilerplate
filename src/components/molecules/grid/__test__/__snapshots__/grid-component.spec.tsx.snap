// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Grid Component Snapshot Testing 1`] = `
Array [
  <div
    class="grid"
    data-testid="grid-container"
  >
    <div
      class="grid__row"
      data-testid="grid-row"
    >
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
    </div>
    <div
      class="grid__row"
      data-testid="grid-row"
    >
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
    </div>
    <div
      class="grid__row"
      data-testid="grid-row"
    >
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
    </div>
  </div>,
]
`;
